py3status/modules/bluetooth.py

```diff
@@ -60,14 +61,23 @@ class Py3status:
     """
     # available configuration parameters
     cache_timeout = 10
-    device_separator = '|'
     format = DEFAULT_FORMAT
     format_device = '{name}'
+    format_separator = '\|'
 
     def post_config_hook(self):
-        # Do deprecation stuff here.  Because of the complexity we can't use
-        # the usual deprecation methods
-        # THIS IS A SPECIAL CASE DO NOT USE AS EXAMPLE CODE.
+        self.format_separator = self.py3.safe_format(self.format_separator)
+
+        # DEPRECATION WARNING. SPECIAL CASE. DO NOT USE THIS AS EXAMPLE.
+        old_separator = getattr(self, 'device_separator', None)
+        if old_separator:
+            old_separator = old_separator.replace('|', '\|')
+            self.format_separator = self.py3.safe_format(old_separator)
+            msg = 'DEPRECATION WARNING: you are using old style configuration '
+            msg += 'parameters you should update to use the new format.'
+            self.py3.log(msg)
```

Comment:
We have the deprecation functionality use that.  Yeah we have had to do some stuff in `post_config_hook()` in some modules but only as tricky. hence `# THIS IS A SPECIAL CASE DO NOT USE AS EXAMPLE CODE.`